### PR TITLE
docs: add more examples for prefer-object-spread

### PR DESCRIPTION
**Pull request by coderaiser**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Updated docs related to `prefer-spread` rule according to tests cases. Added missing examples of [**spread syntax**](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#spread_in_object_literals).

### CODE REVIEW

File: docs/src/rules/prefer-object-spread.md [hunk 1]
```diff
@@ -43,14 +41,13 @@ Examples of **correct** code for this rule:
 ```js
 /*eslint prefer-object-spread: "error"*/
 
-Object.assign(...foo);
+({ ...foo });
 
-// Any Object.assign call without an object literal as the first argument
```
Comment (mdjermanovic) [inline docs/src/rules/prefer-object-spread.md:48]: Can we keep this comment somewhere?

**Review by mdjermanovic — COMMENTED** [review]
Thanks for the PR! I agree that documentation for this rule lacks examples with object spread, which is the point of this rule.

File: docs/src/rules/prefer-object-spread.md [hunk 2]
```diff
@@ -43,14 +41,13 @@ Examples of **correct** code for this rule:
...
-Object.assign(foo, { bar: baz });
+({ ...baz, foo: 'bar' });
```
Comment (coderaiser) [inline docs/src/rules/prefer-object-spread.md:46]:
```suggestion
({ ...baz, foo: 'bar' });

// Any Object.assign call without an object literal as the first argument
Object.assign(foo, { bar: baz });
```

File: docs/src/rules/prefer-object-spread.md (hunk 1 above)
Comment (coderaiser) [reply docs/src/rules/prefer-object-spread.md:48]: Sure! Just updated.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks for contributing!

**Pull request merged**